src/main/java/io/quarkus/registry/app/DatabaseRegistryClient.java

```diff
@@ -121,6 +121,33 @@ public ExtensionCatalog resolveNonPlatformExtensionsCatalog(
         return catalog.build();
     }
 
+    @GET
+    @Path("/extensions/all")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "List all extensions. When an extension has multiple releases, only the most recent will be listed.")
+    public ExtensionCatalog resolveAllExtensionsCatalog() {
+        final ExtensionCatalog.Mutable catalog = ExtensionCatalog.builder();
+        List<ExtensionRelease> allExtensionReleases = ExtensionRelease.listAll();
+
+        // We only want the most recent release for each extension, to avoid returning the whole db on each query
+        Map<String, List<ExtensionRelease>> releasesGroupedByExtension = allExtensionReleases.stream()
+                .collect(Collectors.groupingBy(r -> r.extension.groupId + ":" + r.extension.artifactId));
+        Set<Map.Entry<String, List<ExtensionRelease>>> entries = releasesGroupedByExtension.entrySet();
+
+        // Now sort the lists of extension releases, so we can just pick off the first one
+        entries.forEach(es -> es.getValue().sort((er1, er2) -> er2.versionSortable.compareTo(er1.versionSortable)));
+        // We know every extension has at least one release, because that's why it made it into our map
+        List<ExtensionRelease> mostRecentReleases = entries.stream().map(es -> es.getValue().get(0))
+                .toList();
```

Comment:
Couldn't we use `ExtensionRelease.versionSortable` to load only the latest extension release from the database?

Reply:
Oh yes, that seems like it would be more efficient and cleaner and generally better. I had a hammer called "all extensions" and forgot other hammers called "database queries" were available. And this is why we code review. :) 

I'll try that and update!

Reply:
That's now done. Tests are still passing.